lib/Dialect/Torch/Transforms/ScalarizeShapes.cpp

```diff
@@ -424,6 +457,102 @@ class PropagateAtenSliceTensorPattern
 };
 } // namespace
 
+namespace {
+class PropagateAtenTransposeIntPattern
+    : public OpRewritePattern<AtenTransposeIntOp> {
+public:
+  using OpRewritePattern<AtenTransposeIntOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenTransposeIntOp op,
+                                PatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+    ImplicitLocOpBuilder b(loc, rewriter);
+
+    auto selfTy = cast<BaseTensorType>(op.getSelf().getType());
+    auto resultTy = cast<BaseTensorType>(op.getType());
+    if (!selfTy.areAllSizesKnown() || !resultTy.areAllSizesKnown())
+      return rewriter.notifyMatchFailure(op, "requires static sizes");
+
+    SmallVector<OpFoldResult> elements;
+    if (failed(getListFromTensor(op.getSelf(), elements)))
+      return failure();
+
+    int64_t dim0, dim1;
+    if (!matchPattern(op.getDim0(), m_TorchConstantInt(&dim0)))
+      return failure();
+    if (!matchPattern(op.getDim1(), m_TorchConstantInt(&dim1)))
+      return failure();
+
+    ArrayRef<int64_t> selfSizes = selfTy.getSizes();
+    int64_t rank = selfSizes.size();
+
+    int64_t frontDP = 1, dim0L = 1, midDP = 1, dim1L = 1, backDP = 1;
+    for (int64_t i = 0; i < rank; i++) {
+      if (i < dim0) {
+        frontDP *= selfSizes[i];
+        continue;
+      }
+      if (i == dim0) {
+        dim0L *= selfSizes[i];
```

Comment:
```suggestion
        dim0L = selfSizes[i];
```

& could also be moved out of the loop. Makes it more clear that dim0L isnt accumulating

Reply:
Yeah, I felt like this whole section was a bit ugly. I'll clean it up.